homeassistant/components/device_tracker/automatic.py

```diff
@@ -0,0 +1,154 @@
+"""
+Support for the Automatic platform.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/device_tracker.automatic/
+"""
+from datetime import timedelta
+import logging
+import re
+import threading
+import requests
+
+import voluptuous as vol
+
+from homeassistant.components.device_tracker import PLATFORM_SCHEMA
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=30)
+
+CONF_CLIENT_ID = 'client_id'
+CONF_SECRET = 'secret'
+CONF_DEVICES = 'devices'
+
+SCOPE = 'scope:location scope:vehicle:profile scope:user:profile scope:trip'
+
+ATTR_ACCESS_TOKEN = 'access_token'
+ATTR_RESULTS = 'results'
+ATTR_VEHICLE = 'vehicle'
+ATTR_ENDED_AT = 'ended_at'
+ATTR_END_LOCATION = 'end_location'
+
+URL_AUTHORIZE = 'https://accounts.automatic.com/oauth/access_token/'
+URL_VEHICLES = 'https://api.automatic.com/vehicle/'
+URL_TRIPS = 'https://api.automatic.com/trip/'
+
+_VEHICLE_ID_REGEX = re.compile(
+    (URL_VEHICLES + '(.*)?[/]$').replace('/', r'\/'))
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_CLIENT_ID): vol.Coerce(str),
+    vol.Required(CONF_SECRET): vol.Coerce(str),
+    vol.Required(CONF_USERNAME): vol.Coerce(str),
+    vol.Required(CONF_PASSWORD): vol.Coerce(str),
+    vol.Optional(CONF_DEVICES): [vol.Coerce(str)]
+})
+
+
+def setup_scanner(hass, config: dict, see):
+    """Validate the configuration and return an Automatic scanner."""
+    AutomaticDeviceScanner(config, see)
```

Comment:
It's usually a good idea to check if the connection to the third-party entity is possible and let the setup fails if not. Otherwise the users ends up with a non-working platform.